client/src/component/userProvider.js

```diff
@@ -0,0 +1,45 @@
+import React, { useState, createContext, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+
+export const userContext = createContext();
+
+function UserProvider({ children }) {
+  const [isAuth, setIsAuth] = useState(null);
```

Comment:
I think it's better to use role here because we have three type of users (guest, user, admin)
```suggestion
  const [role, setRole] = useState(guest);
```